tests/mbt/model/ccv.qnt

```diff
@@ -662,6 +848,162 @@ module ccv {
+            Ok(currentState)
+        } else {
+            val vscPacket = consumerState.receivedVscPackets.select(packet => packet.id == vscId)[0]
+            val packet = Slash({
+                validator: consumerNode,
+                id: vscId,
+                downtime: downtime,
+                sendingTime: consumerState.chainState.runningTimestamp,
+                timeoutTime: consumerState.chainState.runningTimestamp + CcvTimeout.get(consumer),
+                valPower: vscPacket.validatorSet.get(consumerNode)
+
+            })
+            val newConsumerState = 
+                commitPacketOnConsumer(consumerState, packet)
+                // add the consumer address to the list of outstanding slashes
+                .with("outstandingDowntime", consumerState.outstandingDowntime.union(Set(consumerNode)))
+            Ok(currentState.with("consumerStates", currentState.consumerStates.set(consumer, newConsumerState)))
+        }
+    }
+
+    // receives a given packet on the provider. The arguments are the consumer chain that sent the packet, and the packet itself.
+    pure def recvPacketOnProvider(currentState: ProtocolState, sender: Chain, packet: Packet): Result = {
+        match packet {
+            | VscMatured(p) => recvVscMaturedPacketOnProvider(currentState, sender, p)
+            | Slash(p) => recvSlashPacketOnProvider(currentState, sender, p)
+            | _ => Err("Do not know how to handle this packet type")
+        }
+    }
+
+    pure def recvSlashPacketOnProvider(currentState: ProtocolState, sender: Chain, packet: SlashPacket): Result = {
+        val providerVal = currentState.providerState.consumerAddrToValidator
+            .getOrElse(sender, Map())
+            // if there is no entry for that validator, there is no key assignment
+            .getOrElse(packet.validator, packet.validator)
+        // if the packet is for double-sign, drop it without changing states
+        if (not(packet.downtime) or 
+        // same if the validator is already jailed: no double-jailing!
+            not(providerVal.in(nonJailedNodes(currentState.providerState)))) {
+            Ok(currentState)
+        } else {
+            val jailEndTime = ProviderBlockTime(currentState) + DowntimeJailTime
+            // map the validator to its address on the provider
+            // (in general, the validator is a consumer address and could be an assigned one)
+
+            val slashFactor = DowntimeSlashPercentage
+            
+            val newProviderState = currentState.providerState
+                .jailUntil(providerVal, jailEndTime)
+                // // slashing is currently not enabled in ICS 
+                // .slash(packet.validator, packet.valPower, slashFactor)
+
+            // if the validator did not have voting power of 0 or was already jailed before, the validator set will change due to the slash
+            val valSetChanged = packet.downtime and 
```

Comment:
`packet.downtime` will always be true as it was checked above in the if condition.